async_downloads/static/js/ws_async_downloads.js

```diff
@@ -0,0 +1,179 @@
+$(function () {
+    const url = $("#async-downloads-script").data("url");
+    function readablePeriod(timeStamp) {
+        //determine the timezone offset the browser applies to Date()
+        const offset = new Date().getTimezoneOffset() * 60000;
+
+        const diff = Date.now() - Date.parse(timeStamp) + offset;
+        function suffix(number) {
+            return (number > 1 ? "s" : "") + " ago";
+        }
+        let temp = diff / 1000;
+        const years = Math.floor(temp / 31536000);
+        if (years) return years + " year" + suffix(years);
+        const days = Math.floor((temp %= 31536000) / 86400);
+        if (days) return days + " day" + suffix(days);
+        const hours = Math.floor((temp %= 86400) / 3600);
+        if (hours) return hours + " hour" + suffix(hours);
+        const minutes = Math.floor((temp %= 3600) / 60);
+        if (minutes) return minutes + " minute" + suffix(minutes);
+        const seconds = Math.floor(temp % 60);
+        if (seconds) return seconds + " second" + suffix(seconds);
+        return "now";
+    }
+
+    function calculateTimeStamps() {
+        $(".download-timestamp").each(function () {
+            const timeStamp = $(this).attr("value");
+            $(this).text(readablePeriod(timeStamp));
+        });
+    }
+
+    function refreshDownloads() {
+        // Do nothing if the downloads div doesn't exist
+        if ($("#async-downloads").length === 0) return;
+        $.ajax({
+            method: "GET",
+            url: url,
+        }).done(function (response) {
+            $("#async-downloads .dropdown-menu").html(response.html);
+            calculateTimeStamps();
+        });
+    }
```

Comment:
Is this AJAX call necessary? Shouldn't we just be opening the websocket and reacting to whatever messages it sends?

Reply:
Can be moved also into WS I leaved it here as a redundancy. If we would have problem with WS, AJAX would load this and we can still use downloads
Do you think that we should fully switch to WS?

Reply:
Hm, I think if you specify `WS_MODE`, then that's what we should use. There shouldn't be any issues with websockets not being supported these days.

If the websocket closes for some reason, I think we should try to recover by attempting to open a new connection again.

Reply:
Ok, I will modify that. I already implemented reconnect with 5 sec delay after broken connection.

Reply:
9e01cc5